components/speedreader/rust/lib/src/readability/src/extractor.rs

```diff
@@ -114,57 +112,94 @@ pub fn extract_dom<S: ::std::hash::BuildHasher>(
     // Append siblings of the new root with related content.
     scorer::append_related_siblings(&mut dom, top_candidate.clone());
 
-    scorer::clean(&mut dom, top_candidate.clone(), url, &title.title, features);
-
-    // Our CSS formats based on id="article".
-    dom::set_attr("id", "article", top_candidate.clone(), true);
-
-    let name = QualName::new(None, ns!(), LocalName::from("h1"));
-    let header = dom.create_element(name, vec![], ElementFlags::default());
-    dom.append(
-        &header,
-        NodeOrText::AppendText(StrTendril::from_str(&title.title).unwrap_or_default()),
+    scorer::clean(
+        &mut dom,
+        top_candidate.clone(),
+        &meta.title.split_whitespace().collect::<HashSet<_>>(),
+        url,
+        features,
     );
 
-    if let Some(first_child) = top_candidate.first_child() {
-        dom.append_before_sibling(&first_child.clone(), NodeOrText::AppendNode(header.clone()));
-    }
+    postprocess(&mut dom, top_candidate.clone(), &meta);
 
     // Calls html5ever::serialize() with IncludeNode for us.
-    let content: String = top_candidate.to_string();
-    Ok(Product {
-        title: title.title,
-        content,
-    })
+    let charset_blob = format!("<meta charset=\"{}\"/>", meta.charset);
+    let content: String = charset_blob + &top_candidate.to_string();
+    Ok(Product { meta, content })
+}
+
+pub fn postprocess(dom: &mut Sink, root: Handle, meta: &Meta) {
```

Comment:
nit: `post_process()` perhaps? Not very important TBH, `postprocess` reads well anyways

Reply:
fixed & force-pushed.